gef.py

```diff
@@ -621,18 +621,100 @@ def __str__(self):
     def is_valid(self):
         return "(bad)" not in self.mnemonic
 
+class MallocStateStruct(object):
+    """GEF representation of malloc_state from https://github.com/bminor/glibc/blob/glibc-2.28/malloc/malloc.c#L1658"""
+    def __init__(self, address):
+        try:
+            self.addr = to_unsigned_long(gdb.parse_and_eval("&{}".format(address)))
+        except gdb.error:
+            malloc_hook_addr = to_unsigned_long(gdb.parse_and_eval("(void *)&__malloc_hook"))
+            unaligned_address = malloc_hook_addr + current_arch.ptrsize
+            self.addr = unaligned_address + (0x20 - unaligned_address%0x20)
+
+        self.num_fastbins = 11 if is_x86_32() else 10
+        self.num_bins = 254
+        self.size_t = cached_lookup_type("size_t")
+        self.int_size = cached_lookup_type("int").sizeof
+
+        if get_libc_version() > (2, 26):
+            self.fastbin_offset = self.int_size*4
+        else:
+            self.fastbin_offset = self.int_size*2
+
+        if not self.size_t:
```

Comment:
Nit but why don't you do this nearer to where you first lookup `size_t`?

Reply:
I think I added more stuff above and it moved ;) Can put it closer